app/controllers/declaration/FiscalInformationController.scala

```diff
@@ -60,29 +60,32 @@ class FiscalInformationController @Inject()(
       .bindFromRequest()
       .fold(
         (formWithErrors: Form[FiscalInformation]) => Future.successful(BadRequest(fiscalInformationPage(mode, itemId, formWithErrors))),
-        formData =>
-          formData.onwardSupplyRelief match {
-            case FiscalInformation.AllowedFiscalInformationAnswers.yes =>
-              updateCacheForYes(itemId, formData) map { _ =>
-                navigator.continueTo(routes.AdditionalFiscalReferencesController.displayPage(mode, itemId))
-              }
-            case FiscalInformation.AllowedFiscalInformationAnswers.no =>
-              updateCacheForNo(itemId, formData) map { _ =>
-                navigator.continueTo(routes.CommodityDetailsController.displayPage(mode, itemId))
-              }
-        }
+        formData => updateCache(itemId, formData).map(_ => redirectToNextPage(mode, itemId, formData))
       )
   }
 
-  //TODO Use one method instead of updateCacheForYes and updateCacheForNo
-  private def updateCacheForYes(itemId: String, updatedFiscalInformation: FiscalInformation)(implicit req: JourneyRequest[AnyContent]): Future[Unit] =
-    updateExportsDeclarationSyncDirect(model => {
-      model.updatedItem(itemId, item => item.copy(fiscalInformation = Some(updatedFiscalInformation)))
-    }).map(_ => ())
+  private def updateCache(itemId: String, updatedFiscalInformation: FiscalInformation)(
+    implicit request: JourneyRequest[AnyContent]
+  ): Future[Option[ExportsDeclaration]] = {
+    val updateModel: ExportsDeclaration => ExportsDeclaration =
+      (model: ExportsDeclaration) =>
+        updatedFiscalInformation.onwardSupplyRelief match {
+          case FiscalInformation.AllowedFiscalInformationAnswers.yes =>
+            model.updatedItem(itemId, item => item.copy(fiscalInformation = Some(updatedFiscalInformation)))
+          case FiscalInformation.AllowedFiscalInformationAnswers.no =>
+            model.updatedItem(itemId, item => item.copy(fiscalInformation = Some(updatedFiscalInformation), additionalFiscalReferencesData = None))
+      }
 
-  private def updateCacheForNo(itemId: String, updatedFiscalInformation: FiscalInformation)(implicit req: JourneyRequest[AnyContent]): Future[Unit] =
-    updateExportsDeclarationSyncDirect(model => {
-      model.updatedItem(itemId, item => item.copy(fiscalInformation = Some(updatedFiscalInformation), additionalFiscalReferencesData = None))
-    }).map(_ => ())
+    updateExportsDeclarationSyncDirect(updateModel)
+  }
 
+  private def redirectToNextPage(mode: Mode, itemId: String, fiscalInformation: FiscalInformation)(
+    implicit request: JourneyRequest[AnyContent]
+  ): Result =
+    fiscalInformation.onwardSupplyRelief match {
+      case FiscalInformation.AllowedFiscalInformationAnswers.yes =>
```

Comment:
What do you  think about import all things from `FiscalInformation` ?

Reply:
Good point.

Reply:
Done